src/oncall/ui/__init__.py

```diff
@@ -75,8 +76,15 @@
 
 
 def index(req, resp):
-    user = req.env.get('beaker.session', {}).get('user')
-    if user is None and LOGIN_REQUIRED:
+    # attempt SSO login first then default to session based login
+    sso_login_success = False
+    user = None
+    if sso_auth_manager and sso_auth_manager.authenticate(req):
+        sso_login_success = True
+        user = sso_auth_manager.authenticate(req)
```

Comment:
Also, can we make this only call authenticate() once?

Reply:
changed